docs/zh_cn/case/backup_mysql_in_juicefs.md

```diff
@@ -0,0 +1,167 @@
+# 基于 JuiceFS 的 MySQL 备份、验证和恢复方案
+
+MySQL 通常包含了公司最重要的产品和用户信息，是公司的核心资产，一旦丢失或者损坏会导致非常严重的损失（甚至破产），一定要做多重备份以确保数据安全。我们将介绍如何使用 JuiceFS 来大大简化 MySQL 的备份、验证和恢复过程，有效保障数据安全。
+
+JuiceFS 非常适合存储 MySQL 的备份数据，因为：
+
+1. 它是 POSIX 兼容的文件系统，可以直接将备份数据写入 JuiceFS，也可以在 JuiceFS 上运行 MySQL，而不需要经过本地磁盘中转（包括验证和恢复数据），这将节省大量时间，通常是数小时到数天，这些时间在数据恢复时是非常宝贵的。
+2. 它支持透明的数据压缩和存储加密，避免了手动压缩和加密的麻烦，有效地降低存储成本和保障数据隐私，同样也能节省大量的备份和恢复时间。
+3. 它支持以目录为单位的快照功能，可以很方便地做备份验证和恢复。
+4. 它支持所有公有云的所有区，可以很方便地做异地备份。
+5. 它的容量是弹性伸缩的，你只需要按实际使用量付费，完全不用做容量规划和扩容。
+
+## 逻辑备份
+
+逻辑备份是将 MySQL 里面的数据以简单的文本格式（最不容易损坏的格式）存储，常用工具包括 mysqldump 和 mydumper。mysqldump 是 MySQL 原生的备份工具，稳定好用，但是比较慢。mydumper 支持多线程导出，速度更快，推荐使用 mydumper。
+
+目前没有增量逻辑备份的工具，理论上可以对比两次全量备份的数据差异，使用 diff 生成一个差异文件作为增量备份保存。
+
+逻辑备份比较慢，如果数据库太大，推荐每个周末或者每个月做一次逻辑备份。如果数据库比较小，也可以每天或者每小时做逻辑备份。对于公有云中已经有备份功能的 RDS，不同产品备份策略不尽相同，也建议你再做一次逻辑备份，以规避 RDS 的备份故障导致的风险。
+
+### 备份
+
+逻辑备份的数据可以直接写到 JuiceFS 挂载目录（假定是 /jfs ）中，比如用 mydumper 导出数据：
+
+```
+mydumper -B <db_name> --outputdir /jfs/<db_name>_data
+```
+
+逻辑备份的文本数据非常容易被压缩，JuiceFS 会默认使用当前最先进的 ZStandard 压缩算法将数据压缩后再保存，有非常好的写入性能（高达 300MB/s），比 mydumper 的压缩功能快很多。
+
+### 验证
+
+定时备份并不意味着高枕无忧，因为备份也可能出问题，验证备份是十分必要的。对于 MySQL 备份而言，备份是否真的可用，只有用备份进行一次恢复才知道。
+
+验证逻辑备份，可以启用一个空的 MySQL 实例进行恢复，并对恢复后的 MySQL做一些完整性检查，比如检查数据量大小等。
+
+### 恢复
+
+从逻辑备份恢复数据时，先启动一个空的 MySQL 实例，再用 myloader 导入数据：
+
+```
+myloader -B <db_name> --directory /jfs/<db_name>_data
+```
+
+## 物理备份
+
+物理备份是指备份 MySQL 实际使用的数据文件，业界通常使用 Percona 公司提供的开源工具 XtraBackup 直接读 MySQL 的数据文件进行备份，而不需要经过 MySQL 进行逻辑解析，会快很多。
+
+XtraBackup 支持全量备份和增量备份，建议在一定周期内两者交替进行。比如每周末做一次全量备份，之后的 6 天内每天做一次增量备份，备份数据保存 4 周以上。
+
+### 备份
+
+用 xtrabackup 做物理备份，推荐对整个实例的所有数据库做备份，而不是指定某些数据库。
+
+给整个 MySQL 实例的数据做一个全量备份
+
+```
+xtrabackup --backup --target-dir=/jfs/base/
+```
+
+之后可以做增量备份以缩减数据量和备份时间，比如：
+
+```
+xtrabackup --backup --incremental-basedir=/jfs/base --target-dir=/jfs/incr1
+```
+
+再之后的增量备份，可以继续以全量备份为基准，也可以以上一次增量备份为基准，比如：
+
+```
+xtrabackup --backup --incremental-basedir=/jfs/base --target-dir=/jfs/incr2
+```
+
+**推荐统一使用全量备份做基准来做增量备份，虽然数据量会大一点，但恢复流程会更简单，不容易出错。**
+
+### 验证
+
+验证增量备份时，需要把增量数据附加到全量备份上，此时可以利用 JuiceFS 的快照功能快速拷贝一个全量备份用于验证，验证完成后再删掉 。
+
+基于刚才创建的全量备份建一个 snapshot，因为在 MySQL 运行的时候会修改数据文件，如果直接运行在备份数据上，就把备份破坏了。
+
+```
+juicefs snapshot /jfs/base /jfs/base_snapshot
```

Comment:
This is only available in hosted version.

Reply:
I will change the case directory to a file named case.md. After that, articles related to scenarios and cases will only add URLs and titles to case.md. After making this change, articles written in the hosted version can also be indexed.

Reply:
Maybe we should not index hosted version docs & articles in community docs. This may cause confusion.

Reply:
> Maybe we should not index hosted version docs & articles in community docs. This may cause confusion.

Agreed. User may confusion with different version.